Bio/Align/sam.py

```diff
@@ -699,7 +699,7 @@ def _read_next_alignment(self, stream):
                     target += seq[:number]
                 seq = target
                 index = self._target_indices[rname]
-                target = copy.deepcopy(self.targets[index])
+                target = self.targets[index].copy(deep=True)
```

Comment:
In this case, as `self.targets[index]` does not have complex attributes, is there actually a difference between `copy.deepcopy` and `... .copy(deep=True)`?

Reply:
The built-in version has to do a lot of extra work searching for all the attributes recursively, and trying to avoid circular references and other kinds of worse-case-scenarios. My main motivation for this method was to help speed up operations in parsers where the types are well known, but I thought it probably was also worth being a public method since a lot of Python module objects have specialized .copy methods

Reply:
I see. In this case, I would prefer to just create the `SeqRecord` from scratch. The reason is that Biopython always had a problem with software bloat. It is easy to add a function, but removing a function if it does not live up to its expectations is a painful process that takes over a year with deprecation warnings and whatnot. So if there is no absolute need for this public method now, let's not add it.

Reply:
Totally understandable. I could also change them into a private methods instead. When I did a search, I saw multiple calls to the built-in deepcopy, so there may be a maintenance benefit to code readability and reducing repeated lines. I’ll try to investigate later today to see if there’s enough cases to make it work it

Reply:
One option is to implement these functions as `__copy__` and `__deepcopy__` (see the Python documentation for these). Then we are not defining new API, but just implementing the API that already exists in Python.

Reply:
I considered that, but __copy__ is supposed to be a fast shallow copy, which I probably can't do any faster while dealing with _RestrictedDict. I think it would also be best to preserve __deepcopy__ for when we need to be absolutely sure something doesn't share any references, like in tests. I can do this probably about twice as fast by using what we know, but it's probably still too slow to use in hot loops